_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp

```diff
@@ -915,10 +915,16 @@ mfxStatus MFXVideoENCODEVP9_HW::GetVideoParam(mfxVideoParam *par)
     return MFX_ERR_NONE;
 }
 
-inline void UpdatePictureHeader(mfxU32 frameLen, mfxU32 frameNum, mfxU8* pPictureHeader, mfxU32 bufferSize)
+inline mfxStatus UpdatePictureHeader(mfxU32 frameLen, mfxU32 frameNum, mfxU8* pPictureHeader, mfxU32 bufferSize)
 {
+    if (bufferSize == 0)
+        return MFX_ERR_MORE_DATA;
+
     mfxU32 ivf_frame_header[3] = {frameLen, frameNum << 1, 0x00000000};
-    memcpy_s(pPictureHeader, bufferSize, ivf_frame_header, sizeof (ivf_frame_header));
+    //std::copy(std::begin(ivf_frame_header),std::end(ivf_frame_header), pPictureHeader);
+    std::fill_n(pPictureHeader, 12, 0);
```

Comment:
no, it is wrong, previous one was correct, you don't assign zeroes, these are different vals